### PR TITLE
Not filter out contradictory nofailover tag

### PR DESCRIPTION
**Pull request by hughcapet**
* Ensure that `nofailover` will always be used if both `nofailover` and
`failover_priority` tags are provided
* Call `_validate_failover_tags` from  `reload_local_configuration()` as well
* Properly check values in the `_validate_failover_tags()`: `nofailover` value should be casted to boolean like it is done when accessed in other places

### CODE REVIEW

File: patroni/config.py [hunk 1]
```diff
@@ -814,10 +815,12 @@ def _validate_failover_tags(self) -> None:
           bedrock source of truth)
         """
         tags = self.get('tags', {})
+        if 'nofailover' not in tags:
+            return
         nofailover_tag = tags.get('nofailover')
```
Comment (CyberDem0n) [inline patroni/config.py:820]:
```suggestion
        nofailover_tag = bool(tags.get('nofailover'))
```

File: patroni/config.py [hunk 2]
```diff
@@ -814,10 +815,12 @@ def _validate_failover_tags(self) -> None:
...
         failover_priority_tag = parse_int(tags.get('failover_priority'))
         if failover_priority_tag is not None \
-                and (nofailover_tag is True and failover_priority_tag > 0
-                     or nofailover_tag is False and failover_priority_tag <= 0):
+                and (bool(nofailover_tag) is True and failover_priority_tag > 0
+                     or bool(nofailover_tag) is False and failover_priority_tag <= 0):
```
Comment (CyberDem0n) [inline patroni/config.py:824]:
```suggestion
                and (nofailover_tag is True and failover_priority_tag > 0
                     or nofailover_tag is False and failover_priority_tag <= 0):
```

File: features/steps/patroni_api.py [hunk 3]
```diff
@@ -128,6 +128,12 @@ def scheduled_restart(context, url, in_seconds, data):
     context.execute_steps(u"""Given I issue a POST request to {0}/restart with {1}""".format(url, json.dumps(data)))
 
 
+@step('I {action:w} {tag:w} tag in {pg_name:w} config')
+def add_bool_tag_to_config(context, action, tag, pg_name):
+    value = True if action == 'set' else False
```
Comment (CyberDem0n) [inline features/steps/patroni_api.py:133]:
```suggestion
    value = action == 'set'
```

File: patroni/config.py (hunk 1 above)
Comment (hughcapet) [reply patroni/config.py:820]: we need the actual value for the log

**Comment by CyberDem0n** [comment]
:+1:

**Comment by hughcapet** [comment]
👍

**Pull request merged**